src/AzureAppConfigurationImpl.ts

```diff
@@ -105,3 +108,23 @@ export class AzureAppConfigurationImpl extends Map<string, unknown> implements A
         return headers;
     }
 }
+
+function validatedSelectors(selectors?: { keyFilter: string, labelFilter?: string }[]) {
+    if (!selectors || selectors.length === 0) {
+        // Default selector: key: *, label: \0
+        return [{ keyFilter: KeyFilter.Any, labelFilter: LabelFilter.Null }];
+    }
+    return selectors.map(selectorCandidate => {
+        const selector = { ...selectorCandidate };
+        if (!selector.keyFilter) {

```

Comment:
Does this check make sure that key is not null or empty?

Reply:
Yes. `keyFilter` is of string type, so the condition here is to ensure `keyFilter` is not `undefined` or `null` or `""` (empty string).

BTW for Javascript language, you actually can pass anything in if you insist, and here it also ensures it's not number `0` etc. But that's already guarded by the type check at compile time for TypeScript, and not the case we need to take care.
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean#boolean_coercion

More info in case you are interested in. See https://stackoverflow.com/a/154068 for various ways to detect empty strings. To me, the most accurate way to translate "is null or empty string" might be as below, but seldomly used because it's complicated. And we have typescript now to guard unexpected inputs.
```js
function isNullOrEmptyString(str) {
  return str === undefined
     || str === null
     || typeof str === "string" &&  str.length === 0
}
```